src/main/java/edu/harvard/iq/dataverse/DatasetVersionServiceBean.java

```diff
@@ -163,59 +162,31 @@ public DatasetVersion findByFriendlyVersionNumber(Long datasetId, String friendl
         } catch (NumberFormatException n) {
             return null;
         }
-
-        if (minorVersionNumber != null) {
-            String queryStr = "SELECT v from DatasetVersion v where v.dataset.id = :datasetId  and v.versionNumber= :majorVersionNumber and v.minorVersionNumber= :minorVersionNumber";
-            DatasetVersion foundDatasetVersion = null;
-            try {
-                Query query = em.createQuery(queryStr);
-                query.setParameter("datasetId", datasetId);
-                query.setParameter("majorVersionNumber", majorVersionNumber);
-                query.setParameter("minorVersionNumber", minorVersionNumber);
-                foundDatasetVersion = (DatasetVersion) query.getSingleResult();
-            } catch (javax.persistence.NoResultException e) {
-                logger.warning("no ds version found: " + datasetId + " " + friendlyVersionNumber);
-                // DO nothing, just return null.
-            }
-            return foundDatasetVersion;
-
-        }
-        
-        if (majorVersionNumber == null && minorVersionNumber == null) {
-
+        if (majorVersionNumber == null) {
+            // Must always have a major version
```

Comment:
not sure I understand this change - can you clarify?

Reply:
@scolapasta  ...I'm not sure any more. I think I was trying to intuit what the FIXME was about and perhaps tried using the submitToDPN api call with a "1" or "2" and didn't get what I expected. All I can see now is that the old code would allow a query with the major version as null since the first query comes before the check at old line 184. That said, I don't see now that the change does much except simplify at the cost of changing what's returned for a major version number only - now you'd get 1.0 versus 1.x with the largest x.
Looking at the code, I don't see any use of this method except in my submit to DPN call, so I can restore the old code, leave it as is, or perhaps split it into find by friendly version and findLatestByMajorVersion(). What would you like?

Reply:
I do think if you only have a major version, it should return the latest (i.e. 1.x) and not 1.0. The example being that we if you have a version 1.x you still have version 1 in the citation. Minor revisions are really meant to represent the latest of that version #.

Reply:
OK - I'll revert this.